config/app/com/microsoft/azure/iotsolutions/uiconfig/webservice/v1/models/PackageListApiModel.java

```diff
@@ -37,8 +39,41 @@ public PackageListApiModel() {
     }
 
     public PackageListApiModel(Iterable<Package> models) {
-        this.items = StreamSupport.stream(models.spliterator(), false)
-                .map(m -> new PackageApiModel(m)).collect(Collectors.toList());
+        this(models, null, null);
+    }
+
+    /**
+     * Filters out packages received from the source by packageType and ConfigType.
+     * @param models Package models received from the source
+     * @param packageType Type of packages to retain
+     * @param configType Config Type of packages to retain
+     */
+    public PackageListApiModel(Iterable<Package> models, String packageType, String configType) {
+
+        if (StringUtils.isBlank(packageType))
```

Comment:
As mentioned in previous comment, these filtering should go to a separate service method

Reply:
done